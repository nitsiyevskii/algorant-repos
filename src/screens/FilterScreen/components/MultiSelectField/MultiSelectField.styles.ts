import { createStyles } from 'src/utils/createStyles';

export const useStyles = createStyles((theme) => ({
  container: {
    marginBottom: 8,
  },
  label: {
    fontSize: theme.fonts.size.lg,
    fontWeight: theme.fonts.weight.semibold,
    color: theme.colors.text.primary,
    marginBottom: 16,
  },
  chipsContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    gap: 8,
  },
}));

