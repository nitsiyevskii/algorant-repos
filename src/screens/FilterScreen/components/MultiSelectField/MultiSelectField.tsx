import React from 'react';
import { View, Text } from 'react-native';
import { Chip } from 'src/components/Chip/Chip';
import { CheckboxItem } from 'src/components/CheckboxItem/CheckboxItem';
import { useStyles } from './MultiSelectField.styles';

export type SelectionType = 'checkbox' | 'chip';

interface MultiSelectFieldProps {
  label: string;
  options: Array<{ label: string; value: string }>;
  selectedValues: string[];
  onToggle: (value: string) => void;
  type?: SelectionType;
}

export const MultiSelectField: React.FC<MultiSelectFieldProps> = ({
  label,
  options,
  selectedValues,
  onToggle,
  type = 'checkbox',
}) => {
  const styles = useStyles();

  const isSelected = (value: string) => selectedValues.includes(value);

  if (type === 'chip') {
    return (
      <View style={styles.container}>
        <Text style={styles.label}>{label}</Text>
        <View style={styles.chipsContainer}>
          {options.map((option) => (
            <Chip
              key={option.value}
              label={option.label}
              isSelected={isSelected(option.value)}
              onPress={() => onToggle(option.value)}
              testID={`chip-${option.value}`}
            />
          ))}
        </View>
      </View>
    );
  }

  return (
    <View style={styles.container}>
      <Text style={styles.label}>{label}</Text>
      {options.map((option) => (
        <CheckboxItem
          key={option.value}
          label={option.label}
          isSelected={isSelected(option.value)}
          onPress={() => onToggle(option.value)}
          testID={`checkbox-${option.value}`}
        />
      ))}
    </View>
  );
};

