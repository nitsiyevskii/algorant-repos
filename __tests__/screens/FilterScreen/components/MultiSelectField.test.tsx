import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { MultiSelectField } from 'src/screens/FilterScreen/components/MultiSelectField/MultiSelectField';

describe('MultiSelectField', () => {
  const mockOnToggle = jest.fn();
  const options = [
    { label: 'Option 1', value: 'option1' },
    { label: 'Option 2', value: 'option2' },
    { label: 'Option 3', value: 'option3' },
  ];

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Checkbox type', () => {
    it('renders correctly with label', () => {
      const { getByText } = render(
        <MultiSelectField
          label="Organizations"
          options={options}
          selectedValues={[]}
          onToggle={mockOnToggle}
          type="checkbox"
        />
      );

      expect(getByText('Organizations')).toBeTruthy();
      expect(getByText('Option 1')).toBeTruthy();
      expect(getByText('Option 2')).toBeTruthy();
      expect(getByText('Option 3')).toBeTruthy();
    });

    it('renders all options', () => {
      const { getByText } = render(
        <MultiSelectField
          label="Test"
          options={options}
          selectedValues={[]}
          onToggle={mockOnToggle}
          type="checkbox"
        />
      );

      options.forEach((option) => {
        expect(getByText(option.label)).toBeTruthy();
      });
    });

    it('calls onToggle when checkbox is pressed', () => {
      const { getByTestId } = render(
        <MultiSelectField
          label="Test"
          options={options}
          selectedValues={[]}
          onToggle={mockOnToggle}
          type="checkbox"
        />
      );

      const checkbox = getByTestId('checkbox-option1');
      fireEvent.press(checkbox);

      expect(mockOnToggle).toHaveBeenCalledWith('option1');
    });

    it('renders selected checkboxes correctly', () => {
      const { toJSON } = render(
        <MultiSelectField
          label="Test"
          options={options}
          selectedValues={['option1', 'option3']}
          onToggle={mockOnToggle}
          type="checkbox"
        />
      );

      expect(toJSON()).toBeTruthy();
    });

    it('matches snapshot with no selection', () => {
      const { toJSON } = render(
        <MultiSelectField
          label="Organizations"
          options={options}
          selectedValues={[]}
          onToggle={mockOnToggle}
          type="checkbox"
        />
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('matches snapshot with selections', () => {
      const { toJSON } = render(
        <MultiSelectField
          label="Organizations"
          options={options}
          selectedValues={['option1', 'option2']}
          onToggle={mockOnToggle}
          type="checkbox"
        />
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Chip type', () => {
    it('renders correctly with label', () => {
      const { getByText } = render(
        <MultiSelectField
          label="Languages"
          options={options}
          selectedValues={[]}
          onToggle={mockOnToggle}
          type="chip"
        />
      );

      expect(getByText('Languages')).toBeTruthy();
      expect(getByText('Option 1')).toBeTruthy();
      expect(getByText('Option 2')).toBeTruthy();
      expect(getByText('Option 3')).toBeTruthy();
    });

    it('renders all options', () => {
      const { getByText } = render(
        <MultiSelectField
          label="Test"
          options={options}
          selectedValues={[]}
          onToggle={mockOnToggle}
          type="chip"
        />
      );

      options.forEach((option) => {
        expect(getByText(option.label)).toBeTruthy();
      });
    });

    it('calls onToggle when chip is pressed', () => {
      const { getByTestId } = render(
        <MultiSelectField
          label="Test"
          options={options}
          selectedValues={[]}
          onToggle={mockOnToggle}
          type="chip"
        />
      );

      const chip = getByTestId('chip-option2');
      fireEvent.press(chip);

      expect(mockOnToggle).toHaveBeenCalledWith('option2');
    });

    it('renders selected chips correctly', () => {
      const { toJSON } = render(
        <MultiSelectField
          label="Test"
          options={options}
          selectedValues={['option2']}
          onToggle={mockOnToggle}
          type="chip"
        />
      );

      expect(toJSON()).toBeTruthy();
    });

    it('matches snapshot with no selection', () => {
      const { toJSON } = render(
        <MultiSelectField
          label="Languages"
          options={options}
          selectedValues={[]}
          onToggle={mockOnToggle}
          type="chip"
        />
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('matches snapshot with selections', () => {
      const { toJSON } = render(
        <MultiSelectField
          label="Languages"
          options={options}
          selectedValues={['option1', 'option3']}
          onToggle={mockOnToggle}
          type="chip"
        />
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Default type', () => {
    it('defaults to checkbox type when type is not specified', () => {
      const { getByTestId } = render(
        <MultiSelectField
          label="Test"
          options={options}
          selectedValues={[]}
          onToggle={mockOnToggle}
        />
      );

      const checkbox = getByTestId('checkbox-option1');
      expect(checkbox).toBeTruthy();
    });
  });
});

