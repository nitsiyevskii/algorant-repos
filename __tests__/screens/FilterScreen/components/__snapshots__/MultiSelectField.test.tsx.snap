// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectField Checkbox type matches snapshot with no selection 1`] = `
<View
  style={
    {
      "marginBottom": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "600",
        "marginBottom": 16,
      }
    }
  >
    Organizations
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
    testID="checkbox-option1"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#007AFF",
          "borderRadius": 6,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 24,
        }
      }
    />
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
    testID="checkbox-option2"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#007AFF",
          "borderRadius": 6,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 24,
        }
      }
    />
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    >
      Option 2
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
    testID="checkbox-option3"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#007AFF",
          "borderRadius": 6,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 24,
        }
      }
    />
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    >
      Option 3
    </Text>
  </View>
</View>
`;

exports[`MultiSelectField Checkbox type matches snapshot with selections 1`] = `
<View
  style={
    {
      "marginBottom": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "600",
        "marginBottom": 16,
      }
    }
  >
    Organizations
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
    testID="checkbox-option1"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#007AFF",
          "borderRadius": 6,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 24,
        }
      }
    >
      <Text
        color="#007AFF"
        name="checkmark"
        size={18}
        testID="icon"
      >
        checkmark
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
    testID="checkbox-option2"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#007AFF",
          "borderRadius": 6,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 24,
        }
      }
    >
      <Text
        color="#007AFF"
        name="checkmark"
        size={18}
        testID="icon"
      >
        checkmark
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    >
      Option 2
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
        "paddingVertical": 12,
      }
    }
    testID="checkbox-option3"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#007AFF",
          "borderRadius": 6,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 24,
        }
      }
    />
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 16,
        }
      }
    >
      Option 3
    </Text>
  </View>
</View>
`;

exports[`MultiSelectField Chip type matches snapshot with no selection 1`] = `
<View
  style={
    {
      "marginBottom": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "600",
        "marginBottom": 16,
      }
    }
  >
    Languages
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "gap": 8,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#E5E5EA",
          "borderColor": "#C6C6C8",
          "borderRadius": 20,
          "borderWidth": 1,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
      testID="chip-option1"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "500",
            },
            false,
          ]
        }
      >
        Option 1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#E5E5EA",
          "borderColor": "#C6C6C8",
          "borderRadius": 20,
          "borderWidth": 1,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
      testID="chip-option2"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "500",
            },
            false,
          ]
        }
      >
        Option 2
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#E5E5EA",
          "borderColor": "#C6C6C8",
          "borderRadius": 20,
          "borderWidth": 1,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
      testID="chip-option3"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "500",
            },
            false,
          ]
        }
      >
        Option 3
      </Text>
    </View>
  </View>
</View>
`;

exports[`MultiSelectField Chip type matches snapshot with selections 1`] = `
<View
  style={
    {
      "marginBottom": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 18,
        "fontWeight": "600",
        "marginBottom": 16,
      }
    }
  >
    Languages
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "gap": 8,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#007AFF",
          "borderColor": "#007AFF",
          "borderRadius": 20,
          "borderWidth": 1,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
      testID="chip-option1"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "500",
            },
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        Option 1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#E5E5EA",
          "borderColor": "#C6C6C8",
          "borderRadius": 20,
          "borderWidth": 1,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
      testID="chip-option2"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "500",
            },
            false,
          ]
        }
      >
        Option 2
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#007AFF",
          "borderColor": "#007AFF",
          "borderRadius": 20,
          "borderWidth": 1,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
      testID="chip-option3"
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontSize": 14,
              "fontWeight": "500",
            },
            {
              "color": "#FFFFFF",
            },
          ]
        }
      >
        Option 3
      </Text>
    </View>
  </View>
</View>
`;
